core/new-gui/src/app/workspace/component/workflow-editor/workflow-editor.component.ts

```diff
@@ -1279,66 +1362,103 @@ export class WorkflowEditorComponent implements AfterViewInit, OnDestroy {
         filter(event => this.interactive)
       )
       .pipe(untilDestroyed(this))
-      .subscribe(() => {
-        // if there is something to paste
-        if (this.copiedOperators.size > 0 || this.copiedGroups.size > 0) {
+      .subscribe(async () => {
+        // by reading from the clipboard, permission needs to be granted
+        // a permission prompt automatically shows up by calling readText()
+        navigator.clipboard.readText().then(text => {
+          try {
+            // convert the JSON string in the system clipboard to a JS Map
+            var operatorsInClipboard: Map<string, any> = new Map(Object.entries(JSON.parse(text)));
+            // check if the fields in a normal serialized string exist after converting the JSON string
+            // if not, throw an error, which is propagated and produces an alert for the user
+            if (
+              !operatorsInClipboard.has("operators") &&
+              !operatorsInClipboard.has("operatorPositions") &&
+              !operatorsInClipboard.has("links") &&
+              !operatorsInClipboard.has("groups") &&
+              !operatorsInClipboard.has("breakpoints") &&
+              !operatorsInClipboard.has("commentBoxes")
+            ) {
+              throw new Error("You haven't copied any operators yet!");
+            }
+          } catch (e) {
+            // if the text in the clipboard is not a JSON object, then it means the user
+            // hasn't copied an operator
+            alert("You haven't copied any operator yet");
+            return;
+          }
+
+          // define the arguments required for actually adding operators and links
           const operatorsAndPositions: { op: OperatorPredicate; pos: Point }[] = [];
-          const links: OperatorLink[] = [];
+          let links: Link[] = [];
           const groups: Group[] = [];
           const positions: Point[] = [];
+          // calling get() will give either the value or undefined
+          // at this point, after checking the existence of fields in the operators in the clipboard,
+          // the fields "links" and "operatorPositions" should exist
+          const linksInClipboard: OperatorLink[] = operatorsInClipboard.get("links") as OperatorLink[];
+          const operatorPositionsInClipboard: OperatorPosition = operatorsInClipboard.get(
+            "operatorPositions"
+          ) as OperatorPosition;
+          // get all the operators from the clipboard, which are already sorted by their layers
+          let copiedOps: OperatorPredicate[] = operatorsInClipboard.get("operators") as OperatorPredicate[];
+
+          let linksCopy: LinkWithID = {};
+          copiedOps.forEach(copiedOperator => {
+            // copyOperator assigns a new randomly generated operator ID to the new operator
+            const newOperator = this.copyOperator(copiedOperator);
+
+            for (let link of linksInClipboard) {
+              if (linksCopy[link.linkID] == undefined) {
```

Comment:
I don't think this step needs to be inside the loop for each operator, this adds unnecessary complexity.

Reply:
resolved when communicating on Slack